openslides_backend/presenter/get_users.py

```diff
@@ -0,0 +1,133 @@
+from operator import itemgetter
+from typing import Any, Dict, List
+
+import fastjsonschema
+
+from ..services.datastore.deleted_models_behaviour import DeletedModelsBehaviour
+from ..shared.exceptions import PresenterException
+from ..shared.patterns import Collection
+from ..shared.schema import schema_version
+from .base import BasePresenter
+from .presenter import register_presenter
+
+get_users_schema = fastjsonschema.compile(
+    {
+        "$schema": schema_version,
+        "type": "object",
+        "title": "get_users",
+        "description": "get user ids data",
+        "properties": {
+            "start_index": {"type": "integer"},
+            "entries": {"type": "integer"},
+            "sort_criteria": {
+                "type": "array",
+                "item": {
+                    "type": "string",
+                    "enum": ["username", "first_name", "last_name"],
+                },
+            },
+            "reverse": {"type": "boolean"},
+            "include_temporary": {"type": "boolean"},
+            "filter": {"type": ["string", "null"]},
+        },
+        "required": [],
+        "additionalProperties": False,
+    }
+)
+
+ALLOWED = {
+    "first_name": "",
+    "last_name": "",
+    "username": "",
+    "title": "",
+    "is_active": False,
+    "is_physical_person": False,
+    "email": "",
+    "last_email_send": "",
+    "is_demo_user": False,
+    "number": "",
+    "structure_level": "",
+    "vote_weight": "",
+}
+
+
+@register_presenter("get_users")
+class GetUsers(BasePresenter):
+    """
+    Gets all users and return some user_ids.
+    """
+
+    schema = get_users_schema
+
+    def get_result(self) -> Any:
+        criteria = self.get_and_check_criteria()
+        users = self.get_all_users(criteria)
+        users = self.filter_temp_users(users)
+        users = self.filter_keyword(users)
+        users = self.sort_users(users, criteria)
+        users = self.paginate_users(users)
+        return {"users": [user["id"] for user in users]}
+
+    def get_and_check_criteria(self) -> List[str]:
+        default_criteria = ["last_name", "first_name", "username"]
+        criteria = self.data.get("sort_criteria", default_criteria)
+
+        not_allowed = [crit for crit in criteria if crit not in ALLOWED]
+        if not_allowed:
+            raise PresenterException(f"Sort criteria '{not_allowed}' are not allowed")
+        return criteria
+
+    def get_all_users(self, criteria: List[str]) -> List[Dict[str, Any]]:
+        fields = criteria[:]
+        for name in ("username", "first_name", "last_name"):
```

Comment:
maybe add `meeting_id` and `id` here, so you can just pass `fields` to `get_all`.

Reply:
Okay.